src/main/java/qna/service/QnAService.java

```diff
@@ -35,22 +35,13 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
+        question.delete(loginUser);
 
         List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
         deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
+
+        List<Answer> answers = question.getAnswers();
         for (Answer answer : answers) {
-            answer.setDeleted(true);
             deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
         }
         deleteHistoryService.saveAll(deleteHistories);
```

Comment:
```suggestion
    Question question = findQuestionById(questionId);
    List<DeleteHistory> deleteHistories = question.delete(loginUser);
    deleteHistoryService.saveAll(deleteHistories);
```

아직 값을 꺼내와서 연산하는 로직들이 있는데요.
해당 객체에 메시지를 보내보는 것은 어떨까요
getter를 사용할 때는 코드스멜로 보는 것도 좋다고 생각합니다.
getter 메서드를 사용하게 되면 캡슐화가 깨질 가능성이 높아집니다.
getter 메서드를 사용함으로써 외부에 public으로 해당 변수가 존재한다는 사실을 공개하게 되는데요.
이렇게 내부 데이터에 초점을 두고 프로그래밍 할 경우에 내부의 변화가 외부로 전파될 경우가 발생하게 됩니다.